pkg/types/builder.go

```diff
@@ -288,16 +408,24 @@ func (g *Builder) buildSchema(sch *schema.Schema, cfg *config.Resource, tfPath [
 			// If it's optional but also computed, then it means the field has a server
 			// side default but user can change it, so it needs to go to parameters.
 			switch {
-			case isObservation(sch):
+			case isObservation(f.Sch):
 				if obsType == nil {
 					return nil, errors.Errorf("element type of %s is computed but the underlying schema does not return observation type", fieldPath(names))
 				}
 				elemType = obsType
+				if paramType.Underlying().String() != emptyStruct {
```

Comment:
I didn't realize this is fixing https://github.com/crossplane/terrajet/issues/27 but I understood it after syncing with @sergenyalcin . Well done for such clean solution! We probably need to drop a comment here though to make it more clear why we're doing this check.